dbt/adapters/bigquery/connections.py

```diff
@@ -491,18 +518,26 @@ def raw_execute(
 
         job_creation_timeout = self.get_job_creation_timeout_seconds(conn)
         job_execution_timeout = self.get_job_execution_timeout_seconds(conn)
+        # build out determinsitic_id
+        model_name = conn.credentials.schema  # schema name as model name is not
+        invocation_id = str(uuid.uuid4())
+        job_id = define_job_id(model_name, invocation_id)
```

Comment:
I don't think `uuid.uuid4()` is deterministic, which means `job_id` is not either. Have you considered an `md5` hash of sufficient attributes (model, connection name, etc.)?

Reply:
currently calling `uuid` directly as part of getting unit tests swapped over for functionality I think initial/current plan was to use the `invocation_id` we define via tracking in core https://docs.getdbt.com/reference/dbt-jinja-functions/invocation_id and it itself is a `uuid` based on docs.

Reply:
Instead of using invocation_id (which we only sometimes have) should we just use the actual query text (which we have to have)?